src/clj/puppetlabs/services/jruby/jruby_metrics_core.clj

```diff
@@ -154,6 +164,85 @@
+  (let [request-info (instance-request-info request)
+        reason (:reason request-info)
+        timer-name (cond
+                     (nil? (schema/check HttpRequestReasonInfo reason))
+                       (-> (get-in reason [:request :route-id])
+                           ;; Remove ":" characters created by stringifying
+                           ;; Clojure keywords.
+                           (str/replace ":" "")
+                           ;; Remove segments created from regexes by Comidi.
+                           ;; Regexes contain characters that create escaping
+                           ;; problems in downstream services that use regexes
+                           ;; to search for or select metrics by name.
+                           (str/replace #"-\/.*\/" ""))
+                     (keyword? reason)
+                       (if-let [name-space (namespace reason)]
+                         (str name-space "." (name reason))
+                         (name reason))
+                     :else "other")]
+    (->> timer-name
+         (str "jruby.borrow-timer.")
+         (metrics/host-metric-name hostname)
+         (.timer metric-registry))))
+
+(schema/defn borrow-timers :- java.util.Map
+  "Returns a map of borrow timers from JRuby Metrics"
+  [{:keys [metric-registry hostname]} :- JRubyMetrics]
+  (let [metric-namespace (metrics/host-metric-name hostname "jruby.borrow-timer")
+        ;; TODO: v4.0 of Dropwizard Metrics has a MetricFilter/startsWith
+        ;;       static method that returns a filter which can be passed
+        ;;       directly to .getTimers.
+        metric-filter (partial filter (fn [[k v]]
+                                        (str/starts-with? k metric-namespace)))]
+    (->> (.getTimers metric-registry)
+         metric-filter
+         (into {}))))
+
+(schema/defn summarize-borrow-timers
+  "Generates a summary for each JRuby borrow timer
+
+  This function iterates through the list of registered JRuby borrow timers
+  and summarizes each one."
+  [{:keys [hostname] :as metrics} :- JRubyMetrics]
+  (let [timer-namespace (metrics/host-metric-name hostname "jruby.borrow-timer")
+        get-name (fn [timer-name]
+                   (if (= timer-namespace timer-name)
+                     "total"
+                     (str/replace-first timer-name
+                                        (str timer-namespace ".") "")))
+        timer-summary (fn [timer]
+                        (let [snapshot (.getSnapshot timer)]
+                        {:count (.getCount timer)
+                         :mean (->> snapshot
+                                   .getMean
+                                   (.toMillis TimeUnit/NANOSECONDS))
+                         :max  (->> snapshot
+                                    ;; Using 99.9th percentile instead of the
+                                    ;; max as it is properly weighted for the
+                                    ;; past 5 minutes.
```

Comment:
I don't understand

Reply:
The metrics library defaults to using an ExponentiallyDecayingReservoir which stores the last 1028 measurements. Statistics generated from this reservoir, such as the `mean` and `rate` are weighted to be representative of the past 5 minutes.

Except for the `max`. Max is just the largest number in the list of 1028 measurements.

So, if we used `max` we'd have two measurements that are weighted to represent the past 5 minutes and one that isn't. 99.9th percentile does use the same weighting, so it can be meaningfully compared with the `mean` and `rate`.